content/posts/clean-your-backlog.md

```diff
@@ -0,0 +1,32 @@
+---
+title: "Clean your backlog"
+date: 2021-11-10
+language: en
+author: Dimiter Petrov
+---
+
+Have you recently checked the size of your backlog?
+
+There is a limit to the amount of work a team can do at a given time, so new tasks get queued in a backlog. As a project grows, so does its backlog. There are new features to develop, more bugs to fix, but these tasks take longer to finish than to write down.
+
+## The problem with large backlogs
+
+Large backlogs are demoralizing. It is intimidating to realize how much is left to do.
+
+Large backlogs are overwhelming. People lose the overview. They create a second, smaller, "prioritized" backlog. Or they start tracking work through other channels: in-person meetings, email, text chat. That exacerbates the problem - now you have multiple backlogs and they all keep growing.
+
+Large backlogs stagnate. The bigger the backlog, the more information is outdated and the more maintenance is required.
+
+Large backlogs give a false sense of how much is really planned, because most items are too vague or not estimated yet.
+
+## Keep your backlog small(er)
+
+Reducing the backlog size can reduce the team's stress level. It also allows the team to focus on what's the most important. Here are some ideas to help you declutter:
+
+- **Acknowledge your backlog is too big.** They all are. Discuss that with your team and agree on what you want to do about it.
+- **Review the backlog as your priorities change.** What you though was important 4 months ago may not be as important now. Outright remove what has become irrelevant and reorder the rest to align with the new requirements.
+- **Drop items from the end of the backlog.** A little controversial, but if you keep the list in order of priority some items will always remain at the bottom. Why keep them at all if you never get around to them?
+- **Don't put technical debt tickets in the backlog.** Sometimes you identify code that would benefit from refactoring. If you can do it right away, just go ahead. If you don't have time, there is a good chance the project will have changed significantly by the time you do. The tech debt ticket will be outdated and you will need to assess the situation anew anyhow. On top of that, outdated tickets get further postponed if they are not actionable. My strategy is to address technical debt in small chunks as I work on a task affected by it, with the understanding that I will need more time.
```

Comment:
I think it's important to track somewhere what areas are affected by technical debt. It gives a rough idea how much maintenance should be done in the project. Also, project members might asses technical debt differently, so I think you should note a common understanding somewhere.

Reply:
I knew there would be some push-back on that point :)

I do understand that some projects also have a big amount of technical debt and that needs to be clearly and regularly communicated to stakeholders. If the place to track/describe technical debt was outside of the backlog, where would it be? A wiki document/google doc? The project readme? "To do" comments in the source code?

Reply:
I think in general the README could be a place. We usually have a section for internal developer notes and having a section there about _hey, there is technical debt, we know it_ could be fine.

Reply:
Why not just place technical dept tickets in another trello column? I dont like the idea that I have to create a PR (to have it in the README) for a techical dept.

Reply:
By the way: I like the post, @crackofdusk. Thanks!

Reply:
thank you for this post, @crackofdusk 
I agree with all of it and see it as encouragement to act on big backlogs.

however, for me it's essential to have a "wall of tech. debt" to make it visible to the customer and developers.

maybe the backlog is the wrong place for tech. debt that is not yet plannable. so why not as @mischasteiner73 suggested, just have a separate place (column, board, ...) 

I suggest to write something like:
```suggestion
- **Don't put technical debt tickets in the backlog.** Sometimes you identify code that would benefit from refactoring. If you can do it right away, just go ahead. If you don't have time, it's good to make it visible. Move technical debt tickets to a dedicated "wall of technical debt" in order to make it visible. Only keep those tickets in the backlog that you will address in the next sprint. All the rest is accepted to be there on the wall. My strategy is to address technical debt in small chunks as I work on a task affected by it, with the understanding that I will need more time.
```

Reply:
@andyundso @tompluess @mischasteiner73 Thanks for you feedback. I find addressing the nuances of handling technical debt would be too distracting for a general post about backlogs, so I will remove the point altogether. I see an opportunity for collaboration on a dedicated post about the topic.

Reply:
See 17e7006